thefuck/rules/switch_lang.py

```diff
@@ -63,15 +77,43 @@ def _switch_command(command, layout):
     return ''.join(_switch(ch, layout) for ch in command.script)
 
 
+def _decompose_korean(command):
+    def _change_double(ch):
+        if ch in DOUBLE_LIST:
+            return DOUBLE_MOD_LIST[DOUBLE_LIST.index(ch)]
+        else:
+            return ch
+
+    hg_str = u''
+    for ch in command.script:
+        if u'가' <= ch <= u'힣':
+            ord_ch = ord(ch) - ord(u'가')
+            hd = ord_ch // 588
+            bd = (ord_ch - 588 * hd) // 28
+            tl = ord_ch - 588 * hd - 28 * bd
+            for ch in [HEAD_LIST[hd], BODY_LIST[bd], TAIL_LIST[tl]]:
+                if ch != ' ':
+                    hg_str += _change_double(ch)
+        else:
+            hg_str += _change_double(ch)
+    return hg_str
+
+
 def match(command):
     if 'not found' not in command.output:
         return False
+    if any([u'ㄱ' <= ch <= u'ㅎ' or u'ㅏ' <= ch <= u'ㅣ' or u'가' <= ch <= u'힣'
```

Comment:
If you remove the square brackets here, you'll be passing a generator to `any` instead of a list. That way, the iteration stops at the very first element of `command.script` that meets the conditional expression, not having to iterate over all of them for every case. Generators FTW! ✨